lib/pages/company_site/eti/table_products.rb

```diff
@@ -0,0 +1,459 @@
+        elements(:copy_product_icon, css: '.js-copy-product')
+
+        # @return элемент строки товара и таблицы Selenium::WebDriver::Element
+        #
+        # @example
+        #   product(name: 'Мяч')
+        #   product(index: 0)
+        #
+        # @param [hash]
+        #  * :name - String
+        #  * :index - Fixnum
+        #
+        def product(param = {})
+          if param.key? :name
+            name_element = names_elements.select { |n| n.text == param[:name] }.first
+            products_elements[names_elements.index(name_element)]
+          elsif param.key? :index
+            products_elements[param[:index]]
+          end
+        end
+
+        # @return элемент строки товара и таблицы Selenium::WebDriver::Element
+        # Поля заполняются в том порядке, в котором переданы в параметрах
+        #
+        # @example
+        #   add_product(name: 'Мяч', exists: :available, group: 'Спортивный инвентарь')
+        #
+        # @param [Hash]
+        #
+        def add_product(params = {})
+          add_new_product
+          wait_until { products_elements[0].attribute('class').include?('new') }
+
+          params.each do |key, value|
+            send("set_#{key}", products_elements[0], value)
+          end
+
+          products_elements[0]
+        end
+
+        def copy_product(product)
+          products = products_elements
+
+          browser
+            .action
+            .move_to(products_elements[product_index(product)].element)
+            .perform
+
+          copy_product_icon_elements[product_index(product)].click
+          wait_until { products_elements.size > products.size }
+        end
+
+        def delete_product(product)
+
+          browser
+            .action
+            .move_to(products_elements[product_index(product)].element)
+            .perform
+
+          confirm(true) { delete_product_icon }
+
+          wait_saving
+        end
+
+        # @return nothing
+        #
+        # @example
+        #   set_name(@product1, 'Футболка')
+        #
+        # @param [Selenium::WebDriver::Element]
+        # @param [String]
+        #
+        def set_name(product, text)
+          click_on_cell(names_elements[product_index(product) || 0])
+
+          browser
+            .action
+            .send_keys(Selenium::WebDriver::Keys::KEYS[:enter])
+            .key_down(:control)
+            .send_keys('a')
+            .key_up(:control)
+            .send_keys(Selenium::WebDriver::Keys::KEYS[:clear])
+            .send_keys(text)
+            .send_keys(Selenium::WebDriver::Keys::KEYS[:enter])
+            .perform
+
+          wait_saving
+        end
+
+        def name(product)
+          names_elements[product_index(product)].element.text
+        end
+
+        def battery(product)
+          {
+            level: battery_level_elements[product_index(product)].text.strip.to_i,
+            title: battery_title_elements[product_index(product)].attribute('title')
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.